Libraries/Azure.psm1

```diff
@@ -596,21 +595,19 @@ Function Delete-ResourceGroup([string]$RGName, [switch]$KeepDisks, [bool]$UseExi
                 Write-LogInfo "Resources in $RGName are deleted."
                 $retValue = $true
             } else {
-                Write-LogWarn "Lock is in place for $RGName. Skipping RG delete!"
-            }
-        }
-        else {
-            Write-LogInfo "Triggering delete operation for Resource Group ${RGName}"
-            $isRgDeleting = Start-DeleteResourceGroup -RGName $RGName
-            if ($isRgDeleting) {
-                Write-LogInfo "Successfully triggered delete operation for Resource Group ${RGName}"
-            } else {
-                Write-LogWarn "Failed to start delete operation for Resource Group ${RGName}"
+                Write-LogInfo "Triggering delete operation for Resource Group ${RGName}"
+                $isRgDeleting = Start-DeleteResourceGroup -RGName $RGName
+                if ($isRgDeleting) {
+                    Write-LogInfo "Successfully triggered delete operation for Resource Group ${RGName}"
+                } else {
+                    Write-LogWarn "Failed to start delete operation for Resource Group ${RGName}"
```

Comment:
Add $retValue with value.

Reply:
https://github.com/LIS/LISAv2/pull/587/files#diff-d6a28ff69ee2b761d72b453db6381cf7R605 is to assign the value to $retValue, do you mean display the $retValue in log?

Reply:
When it hits the line 608, what is the value of $retValue?

Reply:
if not set $retValue, it will be $null, it uses !$retValue to judge if the RG is deleted or not successfully, so $false and $null stands for not delete RG successfully.
I fixed it, we need implement code strictly.